src/test/java/dataloader/JsonLoaderTest.java

```diff
@@ -0,0 +1,28 @@
+package dataloader;
+
+import org.junit.Test;
+import tech.tablesaw.api.Table;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class JsonLoaderTest {
+    /**
+     * Test Scenario: The application wants to read a json dataset
+     * but the String is invalid(empty String).
+     * Expected Output: JsonDataLoader throws an IOException.
+     */
+
+    @Test(expected = IOException.class)
+    public void testEmptyStringForHashMap() throws IOException {
+        JsonDataLoader loader = new JsonDataLoader();
+        loader.parseFile("nonExistent.json", new HashMap<String, Object>());
+    }
+
+    @Test(expected = IOException.class)
```

Comment:
this test won't do what you think it does... it will throw ```NullPointerException```

Reply:
Since you want to test how the function behaves when given a non-existent file, you should pass a non-null ```Table``` instance

Reply:
I thought so too. But I could create a new instance of Table as it has private access to its class. I am getting an error saying Table() has private access to tech.tablesaw.api.Table.
Any ideas?

Reply:
this means thast ```Table``` class is not meant to be instantiated.

Reply:
if its not meant to be instantiated how am I supposed to create the test in the same manner as we did with the first one?